### PR TITLE
[INLONG-5770][DataProxy] Optimize the method of obtaining the local IP address

### LINKED ISSUE #5770: [Improve][DataProxy] Optimize the method of obtaining the local IP address
### Description

Failed to obtain local ip address on some machines

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pocozh**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- [INLONG-5770][DataProxy] Optimize the method of obtaining the local IP address

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5770

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by gosonzhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java [hunk 1]
```diff
@@ -17,59 +17,55 @@
...
-                for (; allInterface.hasMoreElements(); ) {
-                    NetworkInterface oneInterface = allInterface.nextElement();
-                    String interfaceName = oneInterface.getName();
-                    if (oneInterface.isLoopback()
-                            || !oneInterface.isUp()
-                            || !interfaceName
-                            .equalsIgnoreCase(INNER_NETWORK_INTERFACE)) {
-                        continue;
-                    }
+        if (!StringUtils.isEmpty(localIp)) {
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java:42]: Suggest using `StringUtils.isNotBlank()`.

Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java:42]: Suggest using `StringUtils.isNotBlank()`.

File: inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java [hunk 2]
```diff
@@ -17,59 +17,55 @@
...
+            return localIp;
+        }
+        Enumeration<NetworkInterface> allInterface;
+        String ip = "127.0.0.1";
+        try {
+            allInterface = NetworkInterface.getNetworkInterfaces();
+            for (; allInterface.hasMoreElements(); ) {
```
Comment (healchow) [inline inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java:49]: Suggest changing the `for` to `while` or `forEach`.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java (hunk 2 above)
Comment (pocozh) [reply inlong-common/src/main/java/org/apache/inlong/common/util/NetworkUtils.java]: > Suggest changing the `for` to `while` or `forEach`.

done

**Review by healchow — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**